src/genie/libs/parser/iosxe/show_platform.py

```diff
@@ -3391,6 +3392,127 @@ class ShowSwitch(ShowSwitchSchema, ShowSwitchDetail):
+        current_switch = '0'
+        for line in out.splitlines():
+            if line:
+                line = line.strip()
+            else:
+                continue
+            
+            # Mode
+            # Switch mode                  : Virtual Switch
+            # Switch Mode : Standalone
+            m = p1.match(line)
+            if m:
+                group = m.groupdict()
+                parsed_dict["mode"] = group["mode"]
+            
+            # Domain Number
+            # Virtual switch domain number : 123
+            m = p2.match(line)
+            if m:
+                group = m.groupdict()
+                parsed_dict["domain_number"] = int(group["domain_number"])
+            
+            # Local Switch
+            # Local switch number          : 1
+            m = p3_1.match(line)
+            if m:
+                group = m.groupdict()
+                current_switch = str(group["local_switch"])
+                if "switches" not in parsed_dict.keys():
+                    parsed_dict["switches"] = {}
+                if current_switch not in parsed_dict["switches"].keys():
+                    parsed_dict["switches"][current_switch] = {}
+            
+            # Local Switch Role
+            # Local switch operational role: Virtual Switch Active
+            m = p3_2.match(line)
+            if m:
+                group = m.groupdict()
+                if ("switches" in parsed_dict.keys() and
+                    current_switch in parsed_dict["switches"].keys() and
+                    ("role" not in parsed_dict["switches"][current_switch].keys() or
+                     group["local_switch_role"] != parsed_dict["switches"][current_switch]["role"])):
+                    parsed_dict["switches"][current_switch]["role"] = group["local_switch_role"]
+            
+            # Peer Switch
+            # Peer switch number          : 2
+            m = p4_1.match(line)
+            if m:
+                group = m.groupdict()
+                current_switch = str(group["peer_switch"])
+                if "switches" not in parsed_dict.keys():
+                    parsed_dict["switches"] = {}
+                if current_switch not in parsed_dict["switches"].keys():
+                    parsed_dict["switches"][current_switch] = {}
+            
+            # Peer Switch Role
+            # Peer switch operational role: Virtual Switch Standby
+            m = p4_2.match(line)
+            if m:
+                group = m.groupdict()
+                if ("switches" in parsed_dict.keys() and
+                    current_switch in parsed_dict["switches"].keys() and
+                    ("role" not in parsed_dict["switches"][current_switch].keys() or
+                     group["peer_switch_role"] != parsed_dict["switches"][current_switch]["role"])):
+                    parsed_dict["switches"][current_switch]["role"] = group["peer_switch_role"]            
+
+        return {"switch_virtual": parsed_dict }
```

Comment:
This should just return the parsed dict, putting it under the "switch_virtual" key isn't required